### PR TITLE
Kfp workflow for html2parquet

### PR DESCRIPTION
**Pull request by revit13**
## Why are these changes needed?
Add kfp workflow to html2parquet

## Related issue number (if any).
issue #690

### CODE REVIEW

**Review by roytman — COMMENTED** [review]
please run pre-commit with the new files

**Comment by revit13** [comment]
> please run pre-commit with the new files

Done. Thanks

File: kfp/pipeline_generator/single-pipeline/templates/simple_pipeline.py [hunk 1]
```diff
@@ -103,8 +105,13 @@ def {{ pipeline_name }}(
     # Ray cluster
     ray_name: str = "{{ pipeline_name }}-kfp-ray",  # name of Ray cluster
     # Add image_pull_secret and image_pull_policy to ray workers if needed
+    {%- if image_pull_secret != "" %}
```
Comment (roytman) [inline kfp/pipeline_generator/single-pipeline/templates/simple_pipeline.py:108]: can we reuse the same dictionary and just add `image_pull_secret` when it is defined?

File: transforms/add_new_kfp_workflow.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Adding new KFP workflows
+
+This README outlines the steps to add a new KFP workflow for a new transform under [transforms](./transforms/) directory.
+
+1) Create the workflow. [pipeline generator](../kfp/pipeline_generator/single-pipeline/) can be used for that.
+
+2) Create a new `kfp_ray` directory in the transform directory, similar to [this directory](universal/noop/kfp_ray/). Inside this directory, include the following:
```
Comment (roytman) [inline transforms/add_new_kfp_workflow.md]: should we replace the steps: create a directory, and after that add a template and generate the pipeline?

File: transforms/language/html2parquet/kfp_ray/pipeline_definitions.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    s3_access_secret: "s3-secret"
+    image_pull_secret: ""
+    input_folder: "test/html2parquet/input/"
+    output_folder: "test/html2parquet/output/"
+
+pipeline_transform_input_parameters:
+    pipeline_arguments:
+        - name: data_files_to_use
+          type: "str"
+          value: "['.html', '.zip']"
```
Comment (roytman) [inline transforms/language/html2parquet/kfp_ray/pipeline_definitions.yaml:23]: does the transformer support zip files?

Comment (revit13) [reply transforms/language/html2parquet/kfp_ray/pipeline_definitions.yaml:23]: Yes according to https://github.com/IBM/data-prep-kit/blob/dev/transforms/language/html2parquet/python/Makefile#L54

File: transforms/add_new_kfp_workflow.md (hunk 2 above)
Comment (revit13) [reply transforms/add_new_kfp_workflow.md]: Done. Thanks.

**Review by roytman — APPROVED** [review]
LGTM

**Pull request merged**